### PR TITLE
Fix for #3786 OOM on trace_block and trace_replayBlockTransactions RPC methods

### LINKED ISSUE #3786: OutOfMemoryError on JSON RPC method trace_replayBlockTransactions
### Description
As an Besu, I want to get block transactions traces when I call JSON RPC method trace_replayBlockTransactions on blocks with big transactions (> 20M Gas)

### Acceptance Criteria
* trace_replayBlockTransactions JSON RPC call displays block transactions traces

### Steps to Reproduce (Bug)
1. Create a private network of Besu nodes, set for each node these java options -Xms2G -Xmx2G (BESU_OPTS)
2. Send a huge transaction 25M Gas 
3. Call JSON RPC method trace_replayBlockTransactions with the  block number of the transaction sent in 2. as parameter

**Expected behavior:** 
Return the block transactions traces, we can see below the json result template
```json
{
    "jsonrpc": "2.0",
    "id": 1,
    "result":[ 
      { 
        "output":"0x",
        "vmTrace":{
          "code":"0x7f3940be4289e4c3587d88c1856cc95352461992db0a584c281226faefe560b3016000527f14c4d2c102bdeb2354bfc3dc96a95e4512cf3a8461e0560e2272dbf884ef3905601052600851",
          "ops":[ 
            { 
              "cost":3,
              "ex":{ 
                "mem":null,
                "push":[ 
                  "0x8"
                ],
                "store":null,
                "used":16756175
              },
              "pc":72,
              "sub":null
            },
            ...
          ]
        },
        "trace":[ 
          { 
            "action":{ 
              "callType":"call",
              "from":"0xfe3b557e8fb62b89f4916b721be55ceb828dbd73",
              "gas":"0xffadea",
              "input":"0x",
              "to":"0x0100000000000000000000000000000000000000",
              "value":"0x0"
            },
            "result":{ 
              "gasUsed":"0x1e",
              "output":"0x"
            },
            "subtraces":0,
            "traceAddress":[      
            ],
            "type":"call"
          }
        ],
        "stateDiff":{
          "0xfe3b557e8fb62b89f4916b721be55ceb828dbd73":{ 
            "balance":{ 
              "*":{ 
                "from":"0xffffffffffffffffffffffffffffffffc3e12a20b",
                "to":"0xffffffffffffffffffffffffffffffffc3dc5f091"
              }
            },
            "code":"=",
            "nonce":{ 
              "*":{ 
                "from":"0x14",
                "to":"0x15"
              }
            },
            "storage":{              
            }
          }
        },
        "transactionHash":"0x2a5079cc535c429f668f13a7fb9a28bdba6831b5462bd04f781777b332a8fcbd",
      },
      {...}
    ]
}
```
**Actual behavior:** 
java.lang.OutOfMemoryError: Java heap space
```
java.lang.OutOfMemoryError: Java heap space
        at java.base/java.util.Arrays.copyOf(Arrays.java:3689)
        at java.base/java.util.ArrayList.grow(ArrayList.java:238)
        at java.base/java.util.ArrayList.grow(ArrayList.java:243)
        at java.base/java.util.ArrayList.add(ArrayList.java:486)
        at java.base/java.util.ArrayList.add(ArrayList.java:499)
        at org.hyperledger.besu.ethereum.vm.DebugOperationTracer.traceExecution(DebugOperationTracer.java:97)
        at org.hyperledger.besu.evm.EVM.executeNextOperation(EVM.java:81)
        at org.hyperledger.besu.evm.EVM.runToHalt(EVM.java:74)
        at org.hyperledger.besu.evm.processor.AbstractMessageProcessor.codeExecute(AbstractMessageProcessor.java:161)
        at org.hyperledger.besu.evm.processor.AbstractMessageProcessor.process(AbstractMessageProcessor.java:173)
        at org.hyperledger.besu.ethereum.mainnet.MainnetTransactionProcessor.process(MainnetTransactionProcessor.java:485)
        at org.hyperledger.besu.ethereum.mainnet.MainnetTransactionProcessor.processTransaction(MainnetTransactionProcessor.java:397)
        at org.hyperledger.besu.ethereum.mainnet.MainnetTransactionProcessor.processTransaction(MainnetTransactionProcessor.java:181)
        at org.hyperledger.besu.ethereum.api.jsonrpc.internal.processor.BlockTracer.lambda$prepareReplayAction$0(BlockTracer.java:60)
        at org.hyperledger.besu.ethereum.api.jsonrpc.internal.processor.BlockTracer$$Lambda$1375/0x00000008407b5040.performAction(Unknown Source)
        at org.hyperledger.besu.ethereum.api.jsonrpc.internal.processor.BlockReplay.lambda$block$0(BlockReplay.java:60)
        at org.hyperledger.besu.ethereum.api.jsonrpc.internal.processor.BlockReplay$$Lambda$1377/0x00000008407b5840.apply(Unknown Source)
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
        at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1655)
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
        at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
        at org.hyperledger.besu.ethereum.api.jsonrpc.internal.processor.BlockReplay.lambda$block$1(BlockReplay.java:66)
        at org.hyperledger.besu.ethereum.api.jsonrpc.internal.processor.BlockReplay$$Lambda$1376/0x00000008407b5440.perform(Unknown Source)
        at org.hyperledger.besu.ethereum.api.jsonrpc.internal.processor.BlockReplay.performActionWithBlock(BlockReplay.java:155)
        at org.hyperledger.besu.ethereum.api.jsonrpc.internal.processor.BlockReplay.block(BlockReplay.java:52)
        at org.hyperledger.besu.ethereum.api.jsonrpc.internal.processor.BlockTracer.trace(BlockTracer.java:45)
        at org.hyperledger.besu.ethereum.api.jsonrpc.internal.methods.TraceReplayBlockTransactions.traceBlock(TraceReplayBlockTransactions.java:118)
        at org.hyperledger.besu.ethereum.api.jsonrpc.internal.methods.TraceReplayBlockTransactions.lambda$resultByBlockNumber$0(TraceReplayBlockTransactions.java:102)
        at org.hyperledger.besu.ethereum.api.jsonrpc.internal.methods.TraceReplayBlockTransactions$$Lambda$1373/0x00000008407b4840.apply(Unknown Source)
```
**Frequency:** 
100% 

### Versions
* Software version: besu/v22.4.0-RC2-dev-898ce928/linux-x86_64/corretto-java-17
* Java version: openjdk version "17.0.2" 2022-01-18 LTS
* OS Name & Version: NAME="Amazon Linux", VERSION="2", ID="amzn", ID_LIKE="centos rhel fedora", VERSION_ID="2", PRETTY_NAME="Amazon Linux 2", ANSI_COLOR="0;33", CPE_NAME="cpe:2.3: o:amazon:amazon_linux:2", HOME_URL="https://amazonlinux.com/", Amazon Linux release 2 (Karoo)
* Kernel Version: Linux ip-172-31-14-96.us-east-2.compute.internal 4.14.181-140.257.amzn2.x86_64 1 SMP Wed May 27 02:17:36 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
* Cloud VM, type, size: i3.2xlarge

### PR DESCRIPTION
**Pull request by dhallatmaine**
## PR description
This change addresses a situation where all TraceFrame's for all Transactions in a block are loaded into memory, when only the TraceFrame's for a single Transaction are ever needed.

## Fixed Issue(s)
fixes #3786 

## Documentation
- [x] I thought about documentation and added the `doc-change-required` label to this PR if
    [updates are required](https://wiki.hyperledger.org/display/BESU/Documentation).

## Changelog
- [x] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

**Comment by macfarla** [comment]
@dhallatmaine the DCO check is failing - it requires all commits to be signed off - you should be able to follow the instructions https://github.com/hyperledger/besu/pull/5029/checks?check_run_id=11015658751 to fix

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/processor/StreamBlockTrace.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright ConsenSys AG.
```
Comment (macfarla) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/processor/StreamBlockTrace.java:2]:
```suggestion
 * Copyright contributors to Hyperledger Besu.
```

**Review by macfarla — APPROVED** [review]
makes sense to me - thanks for your contrib! just needs DCO fixed :/

**Review by jflo — APPROVED** [review]
LGTM, thank you for your contribution

**Comment by jflo** [comment]
@Gabriel-Trintinalia could you look into the DCO failure on this branch? There are 3 commits that fail DCO check, one is yours and is due to a mismatched email address in the signature.

**Comment by Gabriel-Trintinalia** [comment]
I have cherry-picked the commits of this PR into https://github.com/hyperledger/besu/pull/5062 and fixed the dco issues

**Comment by macfarla** [comment]
closed in favour of #5062

**Pull request closed without merging**